src/controllers/shared.controllers/director.nationality.controller.ts

```diff
@@ -23,21 +23,26 @@ export const getDirectorNationality = async (req: Request, res: Response, next:
     const session: Session = req.session as Session;
     const officerFiling = await getOfficerFiling(session, transactionId, submissionId);
     let currentUrl: string;
+    let directorName: string;
+
     if (isUpdate) {
+      const companyAppointment = await getCompanyAppointmentFullRecord(session, urlUtils.getCompanyNumberFromRequestParams(req), officerFiling.referenceAppointmentId as string);
       currentUrl = urlUtils.getUrlToPath(UPDATE_DIRECTOR_NATIONALITY_PATH, req)
+      directorName = retrieveDirectorNameFromAppointment(companyAppointment)
     } else {
       currentUrl = urlUtils.getUrlToPath(DIRECTOR_NATIONALITY_PATH, req)
+      directorName = retrieveDirectorNameFromFiling(officerFiling)
     }
+
     return res.render(template, {
       templateName: template,
       backLinkUrl: setBackLink(req, officerFiling.checkYourAnswersLink,urlUtils.getUrlToPath(backUrlPath, req)),
       optionalBackLinkUrl: officerFiling.checkYourAnswersLink,
       typeahead_array: NATIONALITY_LIST + "|" + NATIONALITY_LIST + "|" + NATIONALITY_LIST,
       typeahead_value: officerFiling.nationality1 + "|" + officerFiling.nationality2 + "|" + officerFiling.nationality3,
-      directorName: formatTitleCase(retrieveDirectorNameFromFiling(officerFiling)),
+      directorName: formatTitleCase(directorName),
       nationality2_hidden: checkNationality2(officerFiling),
       nationality3_hidden: checkNationality3(officerFiling),
-      isUpdate,
```

Comment:
is isUpdate needed for director-name.html?

Reply:
isUpdate is used on director-name.html for the former names section